extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -0,0 +1,612 @@
+    this.renderedDays_[key] = this.renderPromiseIntoReact_(templatePromise);
+    return this.renderedDays_[key];
+  }
+
+  /**
+   * Returns `true` if a day template exists.
+   * @return {boolean}
+   */
+  hasDayTemplate_() {
+    return this.templates_.hasTemplate(this.element, '[date-template]');
+  }
+
+  /**
+   * @param {!Array<!DateTemplateMapDef>} templates
+   * @param {!moment} date
+   * @return {?Element}
+   */
+  getTemplate_(templates, date) {
+    for (let i = 0; i < templates.length; i++) {
+      if (templates[i].dates.contains(date)) {
+        return templates[i].template;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get the template tag corresponding to a given date.
+   * @param {!moment} date
+   * @return {?Element}
+   */
+  getDayTemplate_(date) {
+    return (
+      this.getTemplate_(this.srcTemplates_, date) ||
+      this.getTemplate_(this.elementTemplates_, date) ||
+      this.srcDefaultTemplate_ ||
+      this.element.querySelector('[date-template][default]')
+    );
+  }
+
+  /**
+   * Render the template that corresponds to the date with its data.
+   * @param {!moment} day
+   * @param {!JsonObject} data
+   */
+  renderDayTemplate_(day, data) {
+    const template = this.getDayTemplate_(day);
+    return this.renderTemplate_(template, data, day.format('D'));
+  }
+
+  /**
+   * Render the info section of the calendar view.
+   */
+  renderInfo() {
+    if (!this.infoTemplatePromise_) {
+      this.infoTemplatePromise_ = this.renderInfoTemplate_();
+    }
+    return this.renderPromiseIntoReact_(this.infoTemplatePromise_);
+  }
+
+  /**
+   * Returns `true` if an info template exists.
+   * @return {boolean}
+   */
+  hasInfoTemplate_() {
+    return this.templates_.hasTemplate(this.element, '[info-template]');
+  }
+
+  /** @return {!Promise<string>} */
+  renderInfoTemplate_() {
+    const template = this.element.querySelector('[info-template]');
+    return this.renderTemplate_(template);
+  }
+
+  /**
+   * Render the given template with the given data. If the template does not
+   * exist, use a fallback string.
+   * The fallback string will be rendered directly into the DOM so it must
+   * not contain unsanitized user-supplied values.
+   * @param {?Element} template
+   * @param {!JsonObject=} opt_data
+   * @param {string=} opt_fallback
+   * @return {!Promise<string>}
+   */
+  renderTemplate_(template, opt_data, opt_fallback) {
+    if (template) {
+      const data = opt_data || /** @type {!JsonObject} */ ({});
+      return this.templates_.renderTemplate(template, data)
+          .then(rendered => {
+            const renderedEvent = createCustomEvent(
+                this.win_,
+                AmpEvents.DOM_UPDATE,
+                /* detail */ null,
+                {bubbles: true});
+            this.container_.dispatchEvent(renderedEvent);
+            return rendered./*REVIEW*/outerHTML;
+          });
+    } else {
+      return Promise.resolve(opt_fallback || '');
```

Comment:
might be worth calling `sanitizeFormattingHtml` from `sanitizer.js` on fallback anyway for additional protection.